### ISSUE TITLE
False positives on Lint/UriRegexp cop

### ISSUE THREAD

**Issue by wata727** [rubocop-hq/rubocop#8466]
RuboCop v0.89.0 reports an offense when using `regexp` method, not `URI.regexp`. A small example is here:

```rb
def regexp
  /\A[a-z ]*\z/
end

'a' =~ regexp
```

## Expected behavior

No offenses found

## Actual behavior

```
Inspecting 1 file
W

Offenses:

example.rb:5:8: W: Lint/UriRegexp: URI.regexp is obsolete and should not be used. Instead, use URI::DEFAULT_PARSER.make_regexp.
'a' =~ regexp
       ^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

Run `rubocop` against the above example code.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.89.0 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.6.6 x86_64-darwin19)
```
